web/libs/core/src/lib/utils/analytics.test.ts

```diff
@@ -0,0 +1,123 @@
+import { registerAnalytics } from "./analytics";
+
+describe("analytics", () => {
+  let originalRequestIdleCallback: any;
+  let originalSendBeacon: any;
+  let mockSendBeacon: jest.Mock;
+
+  beforeEach(() => {
+    // Mock window.requestIdleCallback
+    originalRequestIdleCallback = window.requestIdleCallback;
+    window.requestIdleCallback = (cb: any) => cb();
+
+    // Mock navigator.sendBeacon
+    originalSendBeacon = navigator.sendBeacon;
+    mockSendBeacon = jest.fn(() => true);
+    Object.defineProperty(navigator, "sendBeacon", {
+      value: mockSendBeacon,
+      configurable: true,
+    });
+
+    // Reset APP_SETTINGS before each test
+    (window as any).APP_SETTINGS = {
+      collect_analytics: true,
+    };
+
+    // Mock window.location
+    Object.defineProperty(window, "location", {
+      value: { href: "http://test.com/page" },
```

Comment:
might be good to define `"http://test.com/page"` as a variable and reuse it as opposed to having the string in a number of places